models/User.js

```diff
@@ -1,57 +1,64 @@
-const mongoose = require('mongoose');
-const uniqueValidator = require('mongoose-unique-validator');
-const crypto = require('crypto');
-const {secret} = require("../config");
+import Sequelize from 'sequelize';
+import bcrypt from 'bcryptjs';
 
-const UserSchema = new mongoose.Schema(
-  {
+/**
+ * A model class representing user resource
+ *
+ * @param {Sequelize} sequelize - Sequelize object
+ * @param {Sequelize.DataTypes} DataTypes - A convinient object holding data types
+ * @return {Sequelize.Model} - User model
+ */
+const User = (sequelize, DataTypes) => {
+  /** @type {Sequelize.Model} */
+  const UserSchema = sequelize.define('user', {
     username: {
-      type: String,
-      lowercase: true,
+      type: DataTypes.STRING,
+      allowNull: false,
       unique: true,
-      required: [true, "can't be blank"],
-      match: [/^[a-zA-Z0-9]+$/, 'is invalid'],
-      index: true
+      validate: {
+        is: /^[a-zA-Z0-9]+$/i,
+      },
+      set(val) {
+        this.setDataValue('username', val.toLowerCase()); // convert to lowercase
+      },
     },
     email: {
-      type: String,
-      lowercase: true,
+      type: DataTypes.STRING,
+      allowNull: false,
       unique: true,
-      required: [true, "can't be blank"],
-      match: [/\S+@\S+\.\S+/, 'is invalid'],
-      index: true
+      validate: {
+        isEmail: { msg: 'Must be a valid email address' },
+      },
     },
-    bio: String,
-    image: String,
-    favorites: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Article' }],
-    following: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
-    hash: String,
-    salt: String
-  },
-  { timestamps: true }
-);
-
-UserSchema.plugin(uniqueValidator, { message: 'is already taken.' });
-
-UserSchema.methods.validPassword = function (password) {
-  const hash = crypto
-    .pbkdf2Sync(password, this.salt, 10000, 512, 'sha512')
-    .toString('hex');
-  return this.hash === hash;
-};
-
-UserSchema.methods.setPassword = function (password) {
-  this.salt = crypto.randomBytes(16).toString('hex');
-  this.hash = crypto
-    .pbkdf2Sync(password, this.salt, 10000, 512, 'sha512')
-    .toString('hex');
-};
+    bio: DataTypes.STRING,
+    image: DataTypes.STRING,
+    password: {
+      type: DataTypes.STRING,
+      set(val) {
+        this.setDataValue('password', bcrypt.hashSync(val, 10));
+      }
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      defaultValue: sequelize.NOW,
+    },
+    updatedAt: {
```

Comment:
#### Critical 
- Add a deletedAt timestamp column for when a user is deleted

Reply:
Noted, I'll make provision for that.